examples/background_tasks.py

```diff
@@ -29,32 +29,35 @@ async def on_shutdown(app: web.Application) -> None:
         await ws.close(code=999, message=b"Server shutdown")
 
 
-async def listen_to_redis(app: web.Application) -> None:
-    sub = await aioredis.Redis(host="localhost", port=6379)
-    ch, *_ = await sub.subscribe("news")
+async def listen_to_valkey(app: web.Application) -> None:
+    r = valkey.Valkey(host="localhost", port=6379, encoding="utf-8")
+    ch = r.pubsub()
+    await ch.subscribe("news")
     try:
-        async for msg in ch.iter(encoding="utf-8"):
+        async for msg in ch.listen():
+            if msg["type"] != "message":
+                continue
             # Forward message to all connected websockets:
             for ws in app[websockets]:
-                await ws.send_str(f"{ch.name}: {msg}")
-            print(f"message in {ch.name}: {msg}")
+                await ws.send_str(f"news: {msg}")
+            print(f"message in news: {msg}")
     except asyncio.CancelledError:
         pass
     finally:
-        print("Cancel Redis listener: close connection...")
-        await sub.unsubscribe(ch.name)
-        await sub.quit()
-        print("Redis connection closed.")
+        print("Cancel Valkey listener: close connection...")
+        await ch.unsubscribe()
+        await r.quit()
+        print("Valkey connection closed.")
 
 
 async def start_background_tasks(app: web.Application) -> None:
-    app[redis_listener] = asyncio.create_task(listen_to_redis(app))
+    app[valkey_listener] = asyncio.create_task(listen_to_valkey(app))
 
 
 async def cleanup_background_tasks(app: web.Application) -> None:
     print("cleanup background tasks...")
-    app[redis_listener].cancel()
-    await app[redis_listener]
+    app[valkey_listener].cancel()
+    await app[valkey_listener]
```

Comment:
While we're here, can we change this to a cleanup_ctx, as per:
https://docs.aiohttp.org/en/stable/web_advanced.html#complex-applications